x-pack/plugins/ml/public/components/job_selector/job_selector.js

```diff
@@ -0,0 +1,510 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import React, { useState, useEffect, useRef } from 'react';
+import { PropTypes } from 'prop-types';
+import moment from 'moment';
+
+import { ml } from '../../services/ml_api_service';
+import { JobSelectorTable } from './job_selector_table/';
+import { timefilter } from 'ui/timefilter';
+import { tabColor } from '../../../common/util/group_color_utils';
+import { getGroupsFromJobs, normalizeTimes, setGlobalState } from './job_select_service_utils';
+import { toastNotifications } from 'ui/notify';
+import {
+  EuiBadge,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiLink,
+  EuiSwitch,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+
+export function getBadge({ id, icon, isGroup = false, removeId, numJobs }) {
```

Comment:
This could be just `Badge` or `JobSelectorBadge` (not sure about the name :) ) and be a state functional component.